internal/host/plugin/repository_host_catalog.go

```diff
@@ -321,6 +360,17 @@ func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version
 		return nil, nil, db.NoRowsAffected, errors.Wrap(ctx, err, op)
 	}
 
+	if updateAttributes {
+		if newPlgHc.GetAttributes() != nil {
+			if err := normalizeCatalogAttributes(ctx, plgClient, newPlgHc); err != nil {
+				return nil, nil, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+			}
+			if newCatalog.Attributes, err = proto.Marshal(newPlgHc.GetAttributes()); err != nil {
```

Comment:
I'm a bit confused, are we storing protobuf bytes in attributes here? How is it being used later? How is the type contract communicated to unmarshalers?

Reply:
The bytes are what go into the database; since it's essentially arbitrary data it's stored in marshaled form. The plugin receives a `*structpb.Struct`.